http.go

```diff
@@ -23,6 +23,14 @@ func (s *Server) Handler() http.Handler {
 		w.Write([]byte("ok\n"))
 	})
 
+	mux.HandleFuncC(pat.Get("/builddate"), func(c context.Context, w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(BUILD_DATE))
+	})
+
+	mux.HandleFuncC(pat.Get("/version"), func(c context.Context, w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(BUILD_DATE))
```

Comment:
Should this be `byte(VERSION)` ?

Reply:
Possibly we want a test for this endpoint too? ;)

Reply:
You're right, it should be `VERSION`. 


I'm sort of "meh" on testing it, though. To be honest, even the test we've written for the `/builddate` endpoint is a little overkill - the only things that could cause this test to fail are changes to the build environment, which also would case this test not to run in the first place. So I'm not sure that either test is particularly useful.

Reply:
It would have caught this error ;)

I think it's useful to at least touch this endpoint and ensure that it returns a response and doesn't fail. It's not much farther from there to verifying the response it returns is the expected one. I see your point, but when you think about it from the perspective of "do we have anything that even attempts to ensure that there is even a handler at that path"? I favor having a test

Reply:
> It would have caught this error ;) 

Sure, but so would the monitor :) The point of a regression test is to test regressions in future code, and for something so simple, we don't have a way to write a test that won't have a 100% false negative rate when regressions are introduced.

The reason I'm not inclined to write a test for this is that we *still* have to write the monitor to check for the endpoint, and to check for the "dirty" value. What we need to test here is the behavior of our linker, which we can't really do in-code, by definition - the linker happens outside. 

In all honesty, I probably shouldn't have written the test for the other one, because it gives a false illusion of coverage. If we can't ensure that the code we're testing is the code we'd be running, the test doesn't actually provide value.


(For context, when the Go project migrated to an SSA-based compiler, they had to keep an additional out-of-band test system by way of running two separate compilers simultaneously - any error in the code itself would have immediately manifested in the tests, so no in-band testing would have caught any bugs).

Reply:
Okay, but we should at the very least verify that we have an endpoint that returns successfully, yeah?